src/test/java/apartment/domain/ReservationServiceTest.java

```diff
@@ -0,0 +1,335 @@
+    }
+
+    @Test
+    public void removeReservationWithSpecificNonValidId() throws DataException {
+        assertEquals(false, reservationService.removeReservationById(3, HostRepositoryDouble.HOST));
+    }
+
+    @Test
+    public void removeReservationWillSpecificId() throws DataException {
+        assertEquals(2, reservationService.getReservationForParticularHost(HostRepositoryDouble.HOST).size());
+
+        assertTrue(reservationService.removeReservationById(2, HostRepositoryDouble.HOST));
+
+        assertEquals(1, reservationService.getReservationForParticularHost(HostRepositoryDouble.HOST).size());
+    }
+
+    @Test
+    public void addNullReservation() throws DataException {
+        Result<Reservation> result = reservationService.add(null, false);
+        assertEquals(1, result.getErrorMessages().size());
+
+        assertEquals("Error: Reservation is Null", result.getErrorMessages().get(0));
+    }
+
+    @Test
+    public void addReservationWillNoObjectSet() throws DataException {
+        Reservation reservation = new Reservation();
+        Result<Reservation> result = reservationService.add(reservation, false);
+        assertEquals(5, result.getErrorMessages().size());
+    }
+
+    @Test
+    public void addReservationWithNegativeTotal() throws DataException {
+        Reservation reservation = new Reservation();
+        reservation.setId(3);
+        reservation.setStartDate(LocalDate.now());
+        reservation.setEndDate(LocalDate.now().plusDays(1));
+        reservation.setTotal(new BigDecimal("-2"));
+        reservation.setGuest(GuestRepositoryDouble.GUEST3);
+        reservation.setHost(HostRepositoryDouble.HOST);
+        Result<Reservation> result = reservationService.add(reservation, false);
+        assertEquals(1, result.getErrorMessages().size());
+
+        assertEquals("Error: total is negative this is not possible.", result.getErrorMessages().get(0));
+    }
+
+    @Test
+    public void addReservationWithHostAndGuestThatDoesNotExistInDatabase() throws DataException {
+        Host host = new Host();
+        host.setEmail("yahoo@kdj.com");
+        host.setId("jkldajf-3kjldkafj-dkfjladkfj-kjld");
+        host.setWeekendRate(new BigDecimal("400"));
+
+        Guest basicGuest = new Guest();
+        basicGuest.setPhone("612 09448094");
+        basicGuest.setState("NY");
+        basicGuest.setLastName("basicLast");
+        basicGuest.setFirstName("basicFirst");
+        basicGuest.setEmail("basic-guest@yahoo.com");
+
+        Reservation reservation = new Reservation();
+        reservation.setId(3);
+        reservation.setStartDate(LocalDate.now());
+        reservation.setEndDate(LocalDate.now().plusDays(1));
+        reservation.setTotal(new BigDecimal(400));
+        reservation.setGuest(basicGuest);
+        reservation.setHost(host);
+
+        Result<Reservation> result = reservationService.add(reservation, false);
+        assertEquals(2, result.getErrorMessages().size());
```

Comment:
Always assert that you received the expected error message (to avoid false positives/negatives).